trade-finance-manager-ui/server/controllers/utilisation-reports/index.test.ts

```diff
@@ -65,4 +69,96 @@ describe('controllers/utilisation-reports', () => {
       /* eslint-enable no-underscore-dangle */
     });
   });
+
+  describe('getUtilisationReportByBankId', () => {
+    const consoleErrorSpy = jest.spyOn(console, 'error');
+    const isEmptyMock = jest.fn();
+    const arrayMock = jest.fn();
+    const validationResultMock = () => ({
+      isEmpty: isEmptyMock,
+      array: arrayMock,
+    } as unknown as expressValidator.Result<expressValidator.ValidationError>);
+
+    beforeEach(() => {
+      jest.spyOn(expressValidator, 'validationResult').mockImplementation(validationResultMock);
+    });
+
+    it("should render the 'problem-with-service' page if there are validation errors", async () => {
+      // Arrange
+      const bankId = '123';
+      const { res, req } = httpMocks.createMocks({
+        session: { userToken: 'user-token' },
+        params: { id: bankId },
+      });
+
+      isEmptyMock.mockReturnValueOnce(false);
+      arrayMock.mockReturnValueOnce([]);
+
+      // Act
+      await getUtilisationReportByBankId(req, res);
+
+      // Assert
+      /* eslint-disable no-underscore-dangle */
+      expect(res._getRenderView()).toEqual('_partials/problem-with-service.njk');
+      /* eslint-enable no-underscore-dangle */
+      expect(consoleErrorSpy).toHaveBeenLastCalledWith(`Error rendering utilisation for bank with id ${bankId}:`, []);
+    });
+
+    it("should render the 'problem-with-service' page if the bank id cannot be found", async () => {
+      // Arrange
+      const bankId = '123';
+      const { res, req } = httpMocks.createMocks({
+        session: { userToken: 'user-token' },
+        params: { id: bankId },
+      });
+
+      isEmptyMock.mockReturnValueOnce(true);
+
+      mocked(api.getUtilisationReportsReconciliationSummary).mockResolvedValue([{ bank: { id: '321', name: 'Test bank' } }] as UtilisationReportReconciliationSummaryItem[]);
+
+      // Act
+      await getUtilisationReportByBankId(req, res);
+
+      // Assert
+      /* eslint-disable no-underscore-dangle */
+      expect(res._getRenderView()).toEqual('_partials/problem-with-service.njk');
+      /* eslint-enable no-underscore-dangle */
+      expect(consoleErrorSpy).toHaveBeenLastCalledWith(`Error rendering utilisation for bank with id ${bankId}:`, new Error(`Bank with id ${bankId} not found`));
+    });
+
+    it('should render the correct page with the correct data', async () => {
+      // Arrange
+      const bankId = '123';
+      const { res, req } = httpMocks.createMocks({
+        session: { userToken: 'user-token' },
+        params: { id: bankId },
+      });
+
+      isEmptyMock.mockReturnValueOnce(true);
+
+      const bank = {
+        id: bankId,
+        name: 'Test Bank',
+      };
+      mocked(api.getUtilisationReportsReconciliationSummary).mockResolvedValue([{ bank }] as UtilisationReportReconciliationSummaryItem[]);
```

Comment:
Can use one of the values from `MOCK_UTILISATION_REPORT_RECONCILIATION_SUMMARY_ITEMS` here to get a more realistic result, rather than casting a partial object.